polyfill/lib/ecmascript.mjs

```diff
@@ -1445,9 +1445,13 @@ export const ES = ObjectAssign({}, ES2020, {
   },
   CalendarFields: (calendar, fieldNames) => {
     let fields = calendar.fields;
-    if (fields === undefined) fields = GetIntrinsic('%Temporal.Calendar.prototype.fields%');
-    const array = ES.Call(fields, calendar, [fieldNames]);
-    return ES.CreateListFromArrayLike(array, ['String']);
+    if (fields !== undefined) fieldNames = ES.Call(fields, calendar, [fieldNames]);
+    return ES.CreateListFromArrayLike(fieldNames, ['String']);
+  },
+  CalendarMergeFields: (calendar, fields, additionalFields) => {
+    let mergeFields = calendar.mergeFields;
+    if (mergeFields === undefined) return ObjectAssign(fields, additionalFields);
```

Comment:
Re: Justin's later comment about `monthCode`:

If we add `monthCode` where `month` is requested, then the base implementation of `MergeFields` needs to process `month` <--> `monthCode` conflicts for the ISO calendar; `Object.assign` wouldn't be sufficient in that case, and undefined `mergeFields` as `Object.assign` seems a potential source of bugs. I'm not sure if this means we don't default in `ES.CalendarMergeFields` or `Calendar.prototype.mergeFields`, or something else.

Reply:
I wasn't taking `monthCode` into account in the polyfill since it's not implemented there yet. You're right that once `monthCode` is implemented, the default mergeFields can no longer be Object.assign. Once that happens, I'll split out a separate `ES.ISOMergeFields()` like it is done in the spec text, and call that here.